core/tests/protractor_utils/ProfilePage.js

```diff
@@ -70,10 +80,15 @@ var ProfilePage = function() {
   };
 
   this.expectUserToHaveInterestPlaceholder = async function(expectedText) {
+    await waitFor.visibilityOf(
+      interestPlaceholder,
+      'interest Placeholder taking too long to appear');
     expect(await interestPlaceholder.getText()).toMatch(expectedText);
   };
 
   this.expectUserToNotHaveInterestPlaceholder = async function() {
+    await waitFor.visibilityOf(interestPlaceholder,
+      'interestPlaceholder label element taking too long to appear');
     expect(await interestPlaceholder.isPresent()).toBe(false);
   };
 
```

Comment:
Sorry if my last comment was unclear: we should keep the waitFor you add in `expectToHaveExplorationCards`, just use `.first()` like you did in `expectToHaveExplorationCardByName`